### DISCUSSION TITLE
linkerd-init violates the baseline Pod Security Standard

### DISCUSSION THREAD

**Discussion by MarkKharitonov** [linkerd/linkerd2#11097] [General] [answered]
Enforcing the baseline Pod Security Standard with the linkerd-init sidecar fails with the following error:
```
Error from server (Forbidden): pods "test" is forbidden: violates PodSecurity "baseline:v1.25": non-default capabilities (container "linkerd-init" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add)
```
I wonder what can be done about it? On one hand, linkerd-init must have these capabilities. On the other hand, I do not see how this can reconcile with the PSS. The currently implemented exemptions mechanism (through user name, runtime class or namespace) is not fine enough to apply to a sidecar.

Any ideas on how this conundrum can be resolved?

**Comment by alpeb** [comment, marked as answer]
That's right, unfortunately, linkerd-init is incompatible with a anything but a `privileged` PSS, given that mechanism's coarse grained policies. If you're required to enforce a more restrictive PSS policy you can via the linkerd-cni model, which refrains from using the linkerd-init container for setting up the iptable rules required by the proxy.

**Comment by MarkKharitonov** [comment]
Would you be able to elaborate on the difference between **linkerd-cni** model and **linkerd-init** model? Or should I ask a separate question for that?

**Reply by alpeb** [reply]
There's a [section in the docs](https://linkerd.io/2.13/features/cni/) with some details about it.
